### PR TITLE
Removing python2.6 workarounds

### PR DESCRIPTION
**Pull request by bsipocz**

### CODE REVIEW

**Comment by bsipocz** [comment]
@taldcroft - Could you have a suggestion to revise this comment about python 3.1 and 3.2? https://github.com/astropy/astropy/blob/master/astropy/io/ascii/core.py#L144

I've removed that workaround looking subclassing, but that totally messed up all versions we support (2.7 & 3.4+)

File: astropy/time/tests/test_basic.py [hunk 1]
```diff
@@ -822,12 +822,8 @@ def test_now():
...
 
-    # py < 2.7 doesn't have `total_seconds`
-    if sys.version_info[:2] < (2, 7):
-        total_secs = lambda td: (td.microseconds + (
-            td.seconds + td.days * 24 * 3600) * 10 ** 6) / 10 ** 6.
-    else:
-        total_secs = lambda td: td.total_seconds()
+    total_secs = lambda td: td.total_seconds()
+
     assert total_secs(dt) < 0.1
```
Comment (mhvk) [inline astropy/time/tests/test_basic.py]: I think we can even just make this
```
assert td.total_seconds(dt) < 0.1
```

**Comment by bsipocz** [comment]
@mhvk - Fixed. With 150K+ lines in astropy, no wonder a few of these always slip through.

**Comment by taldcroft** [comment]
@bsipocz - the code comment is wrong, and the underlying issue is that `np.ma.masked` is not (yet) hashable, though I did argue again that it should.  So I would suggest to just change the comment to be:
```
The constant ``numpy.ma.masked`` is not hashable (see https://github.com/numpy/numpy/issues/4660), so we need to 
extend it here with a hash value.
```

**Pull request merged**